src/hooks/useAuthFlow.ts

```diff
@@ -15,7 +14,30 @@ export const useAuthFlow = () => {
   const { url: redirectTo } = useAfterLoginRedirection()
   const { flags, initialized: flagInitialized } = useContext(FeatureFlagsContext)
   const [targetConfig] = useTargetConfig()
-  const [isLoadingUserAgentData, userAgentData] = useAdvancedUserAgentData()
+  const [hasWebGPU, setHasWebGPU] = useState(false)
+  const [isCheckingWebGPU, setIsCheckingWebGPU] = useState(true)
+
+  useEffect(() => {
+    async function checkWebGpu() {
+      setIsCheckingWebGPU(true)
+
+      if (!('gpu' in navigator)) {
+        setHasWebGPU(false)
+        setIsCheckingWebGPU(false)
+        return
+      }
+
+      try {
+        const adapter = await (navigator as unknown as { gpu?: { requestAdapter(): Promise<unknown> } }).gpu?.requestAdapter()
```

Comment:
If I'm not wrong, this check should be quite fast, so, instead of making it a hook, what do you think about just creating an async function that is checked whenever we need to?